ros_buildfarm/templates/devel/devel_task.Dockerfile.em

```diff
@@ -61,10 +61,12 @@ RUN echo "@today_str"
 RUN python3 -u /tmp/wrapper_scripts/apt.py update-install-clean -q -y git python3-yaml
 
 @[if build_tool == 'colcon']@
-RUN python3 -u /tmp/wrapper_scripts/apt.py update-install-clean -q -y python3-pip
 @# colcon-core.package_identification.python needs at least setuptools 30.3.0
 @# pytest-rerunfailures enables usage of --retest-until-pass
-RUN pip3 install -U setuptools==59.6.0 pytest-rerunfailures
+@# using a regex to install python3-pytest-rerunfailures like this effectively makes it optional.
+RUN python3 -u /tmp/wrapper_scripts/apt.py update-install-clean -q -y '^python3-(pip|pytest-rerunfailures|setuptools)$'
+@# use pip to upgrade if necessary
+RUN pip3 install 'setuptools>=30.3.0,<60.0.0' pytest-rerunfailures
```

Comment:
I don't think I totally understand what is going to happen with this sequence of commands.  Let's take the packages one at a time:

* For setuptools, we are always going to succeed installing the debian package for it.  In that case, doing `pip3 install setuptools>=30.3.0,<60.0.0` isn't going to do anything since we aren't using the `-U` flag.  So I think we could just drop installing `setuptools` via `pip` completely.
* For `pytest-rerunfailures`, I don't understand the comment "using a regex to install python3-pytest-rerunfailures like this effectively makes it optional".  The fact that we are specifying it on the command line to `apt.py` means it will attempt to (and fail to) be installed, no?  Maybe I just don't understand what apt.py does.

Reply:
> So I think we could just drop installing `setuptools` via `pip` completely.

As far as Bionic, Focal, and Jammy are concerned, yes. Xenial needs it, and I didn't check Debian, but I'm sure some there need it. Personally, I think it's worth leaving in for now for wider compatibility.

> I don't understand the comment...

Apt doesn't know how many packages will match that regex. It will only produce an error if NO packages match that regex. So on Bionic and Focal, the regex only matches 2 packages, and the command succeeds. On Jammy, it happens to match 3.